code/game/machinery/kitchen/cooking_machines/fryer.dm

```diff
@@ -8,27 +8,165 @@
+	if (..())
+		//Set temperature of oil reagent
+		var/datum/reagent/nutriment/triglyceride/oil/OL = oil.get_master_reagent()
+		if (OL && istype(OL))
+			OL.data["temperature"] = temperature
+
+/obj/machinery/appliance/cooker/fryer/equalize_temperature()
+	if (..())
+		//Set temperature of oil reagent
+		var/datum/reagent/nutriment/triglyceride/oil/OL = oil.get_master_reagent()
+		if (OL && istype(OL))
+			OL.data["temperature"] = temperature
 
-	user.visible_message("<span class='danger'>\The [user] starts pushing \the [victim] into \the [src]!</span>")
-	icon_state = on_icon
-	cooking = 1
 
-	if(!do_mob(user, victim, 20))
-		cooking = 0
+/obj/machinery/appliance/cooker/fryer/update_cooking_power()
+	..()//In addition to parent temperature calculation
+	//Fryer efficiency also drops when oil levels arent optimal
+	var/oil_level = 0
+	var/datum/reagent/nutriment/triglyceride/oil/OL = oil.get_master_reagent()
+	if (OL && istype(OL))
+		oil_level = OL.volume
+
+	var/oil_efficiency = 0
+	if (oil_level)
+		oil_efficiency = oil_level / optimal_oil
+
+		if (oil_efficiency > 1)
+			//We're above optimal, efficiency goes down as we pass too much over it
+			oil_efficiency = 1 - (oil_efficiency - 1)
+
+
+	cooking_power *= oil_efficiency
+
+
+/obj/machinery/appliance/cooker/fryer/update_icon()
+	if (cooking)
+		icon_state = on_icon
+	else
 		icon_state = off_icon
+	..()
+
+
+//Fryer gradually infuses any cooked food with oil. Moar calories
+//This causes a slow drop in oil levels, encouraging refill after extended use
+/obj/machinery/appliance/cooker/fryer/do_cooking_tick(var/datum/cooking_item/CI)
+	if(..() && (CI.oil < CI.max_oil) && prob(20))
+		var/datum/reagents/buffer = new /datum/reagents(2)
+		oil.trans_to_holder(buffer, min(0.5, CI.max_oil - CI.oil))
+		CI.oil += buffer.total_volume
+		CI.container.soak_reagent(buffer)
+
+
+//To solve any odd logic problems with results having oil as part of their compiletime ingredients.
+//Upon finishing a recipe the fryer will analyse any oils in the result, and replace them with our oil
+//As well as capping the total to the max oil
+/obj/machinery/appliance/cooker/fryer/finish_cooking(var/datum/cooking_item/CI)
+	..()
+	var/total_oil = 0
+	var/total_our_oil = 0
+	var/total_removed = 0
+	var/datum/reagent/our_oil = oil.get_master_reagent()
+
+	for (var/obj/item/I in CI.container)
+		if (I.reagents && I.reagents.total_volume)
+			for (var/datum/reagent/R in I.reagents.reagent_list)
+				if (istype(R, /datum/reagent/nutriment/triglyceride/oil))
+					total_oil += R.volume
+					if (R.id != our_oil.id)
+						total_removed += R.volume
+						I.reagents.remove_reagent(R.id, R.volume)
+					else
+						total_our_oil += R.volume
+
+
+	if (total_removed > 0 || total_oil != CI.max_oil)
+		total_oil = min(total_oil, CI.max_oil)
+
+		if (total_our_oil < total_oil)
+			//If we have less than the combined total, then top up from our reservoir
+			var/datum/reagents/buffer = new /datum/reagents(999999999999)
```

Comment:
it should be infinity. its a temporary buffer, it needs to hold whatever it needs to hold